### PR TITLE
Setup leaderboard, match creation and match history

### PR DESCRIPTION
**Pull request by MathijsR94**

### CODE REVIEW

File: src/pages/index.tsx [hunk 1]
```diff
@@ -1,11 +1,58 @@
...
+import styles from './Home.module.css';
+
+const cx = classNames.bind(styles);
+
+const levels = Array.from({ length: 10 }, (_, i) => i + 1);
+
+export default function Home({
+  usersPerLevel,
+}: InferGetServerSidePropsType<typeof getServerSideProps>) {
+  const router = useRouter();
```
Comment (brdv) [inline src/pages/index.tsx]: unused

File: src/components/Layout/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,71 @@
+import classNames from 'classnames/bind';
+import Head from 'next/head';
```
Comment (brdv) [inline src/components/Layout/Layout.tsx]: unused

File: src/components/Layout/Layout.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import { useRouter } from 'next/router';
+import { PrimeReactContext } from 'primereact/api';
+import { Button } from 'primereact/button';
+import {
+  Fragment,
+  PropsWithChildren,
+  useContext,
+  useEffect,
```
Comment (brdv) [inline src/components/Layout/Layout.tsx]: unused

File: src/pages/match/create.tsx [hunk 4]
```diff
@@ -1,11 +1,165 @@
+import { User } from '@prisma/client';
+import classNames from 'classnames/bind';
+import type {
+  GetServerSideProps,
```
Comment (brdv) [inline src/pages/match/create.tsx]: unused

File: src/pages/matches/index.tsx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+import { getUrl } from '@blaco/utils/getUrl';
+import styles from './MatchPage.module.css';
+
+const cx = classNames.bind(styles);
+
+type TMatchInfo = Match & { challenger: User; defender: User };
+
+export default function MatchesPage({
+  matches,
+  users,
```
Comment (brdv) [inline src/pages/matches/index.tsx]: unused

File: src/utils/getAvatarUrl.ts [hunk 6]
```diff
@@ -0,0 +1,3 @@
+export function getAvatarUrl(name: string) {
+  return `/images/users/${name}.jpeg`;
+}
```
Comment (brdv) [inline src/utils/getAvatarUrl.ts:3]: Could you add a check here to see if the file exists, and if not, return 'unnamed.jpeg'? In that way, the images will load smoother and we won't get a bunch of `404: File not found` errors if something is missing. 😁

Comment (MathijsR94) [reply src/utils/getAvatarUrl.ts:3]: I am not sure if this is possible, maybe on build time? So we would need to generate a list of available images and read it on runtime. I've rewritten the UserAvatar a bit and hope this fixes some of the jankiness :) What do you think?

**Review by brdv — APPROVED** [review]

**Pull request merged**